tests/test_dataManager.py

```diff
@@ -112,15 +126,17 @@ def test_call(self):
             output = manager(self.data)
             expected_path = os.path.join(path, "TestBlue_1999-01-01T00:00:00.000.fits")
             self.assertEqual(output, expected_path)
-            hdulist = astropy.io.fits.open(output, checksum=True)
-            self.check_header(hdulist[0].header)
-            np.testing.assert_array_equal(hdulist[0].data, self.spectrum)
-            np.testing.assert_array_equal(Table.read(hdulist[1])['wavelength'], self.wavelength)
-            # Ensure the checksums are written, but let astropy verify them.
-            self.assertIn('CHECKSUM', hdulist[0].header)
-            self.assertIn('DATASUM', hdulist[0].header)
-            self.assertIn('CHECKSUM', hdulist[1].header)
-            self.assertIn('DATASUM', hdulist[1].header)
+            with astropy.io.fits.open(output, checksum=True) as hdulist:
+                self.check_header(hdulist[0].header)
+                np.testing.assert_array_equal(hdulist[0].data, self.spectrum)
+                # Want first row
+                wavelengths = QTable.read(hdulist[1])['wavelength'][0]
```

Comment:
It look like this code is not testing that the FITS WAVE-TAB conventions of having keywords in the image HDU match the EXTNAME and column name of/in the wavelength HDU.

I think it would be better to add a new test function to the existing test program, one which either takes a file path string as input, or perhaps an Astropy HDUList object, and then simply verifies that the associated FITS file follows the WAVE-TAB convention.  This should be separated from the test that shows that the correct wavelength data are read back.